server/internal/usecase/interactor/workspace.go

```diff
@@ -201,6 +201,14 @@ func (i *Workspace) Remove(ctx context.Context, id id.WorkspaceID, operator *use
 			return interfaces.ErrOperationDenied
 		}
 
+		projectCount, err := i.repos.Project.CountByWorkspace(ctx, id)
+		if err != nil {
+			return err
+		}
+		if projectCount > 0 {
+			return user.ErrWorkspaceWithProjects
```

Comment:
```suggestion
			return interfaces.ErrWorkspaceWithProjects
```

This error is not a user-related error. Define it at `internal/usecase/interfaces/workspace.go`